### PR TITLE
*: add log and metric in raft_client 

### PR DESCRIPTION
**Pull request by crazycs520**
### What is changed and how it works?

Issue Number: ref #12362

What's Changed:

```commit-message
add log and metric in raft_client.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by Connor1996 — COMMENTED** [review]
Please update grafana template

File: src/server/raft_client.rs [hunk 1]
```diff
@@ -862,7 +864,11 @@ async fn start<S, R>(
                 .report_store_unreachable(back_end.store_id);
             continue;
         } else {
-            debug!("connection established"; "store_id" => back_end.store_id, "addr" => %addr);
+            let wait_conn_duration = begin.elapsed();
+            info!("connection established"; "store_id" => back_end.store_id, "addr" => %addr, "cost" => ?wait_conn_duration, "msg-count" => ?back_end.queue.len());
```
Comment (overvenus) [inline src/server/raft_client.rs:868]:
```suggestion
            info!("connection established";
                "store_id" => back_end.store_id,
                "addr" => %addr,
                "cost" => ?wait_conn_duration,
                "msg_count" => ?back_end.queue.len(),
            );
```

**Review by overvenus — COMMENTED** [review]
Try keep line width within 80 chars.

**Comment by crazycs520** [comment]
> Please update grafana template

done, PTAL

image

related metrics is:
image

**Review by overvenus — APPROVED** [review]

File: src/server/raft_client.rs [hunk 2]
```diff
@@ -862,7 +864,16 @@ async fn start<S, R>(
                 .report_store_unreachable(back_end.store_id);
             continue;
         } else {
-            debug!("connection established"; "store_id" => back_end.store_id, "addr" => %addr);
+            let wait_conn_duration = begin.elapsed();
```
Comment (cfzjywxk) [inline src/server/raft_client.rs:867]: If the connection is estabilished after serveral round "wait connect timeout", this kind of wait seems not to be recorded, should we add them or add some "wait connect timeout" counter too?

Comment (crazycs520) [reply src/server/raft_client.rs]: nice catch, done.

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Pull request merged**